pkg/helpers/apimachinery_status_conditions.go

```diff
@@ -0,0 +1,29 @@
+package helpers
+
+import (
+	"encoding/json"
+	"sort"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CopyConditions copies the set of conditions
+func CopyConditions(conditions []metav1.Condition) []metav1.Condition {
+	newConditions := make([]metav1.Condition, 0)
```

Comment:
does it make sense to size the new slice to `len(conditions)`?

Reply:
also I believe there is a built in `copy` function

Reply:
> does it make sense to size the new slice to len(conditions)?

Yes... and no :) generally speaking, if we know in advance the length of the source array, we should specify the capacity (not the initial size, as it should be 0). This is to avoid dynamic resizing and memory allocation expensive ops as items are being added. However, we know in advance that there is only one condition (and in  future maybe few more) so, basically, it does not matter as resizing will not happen.

To follow good patterns, I will pre-allocate the destination capacity with the source array length.

Reply:
> also I believe there is a built in copy function

Haven't found one. Happy to hear from an implemented one

I changed the shallow copy I was doing for a deep copy

Reply:
I see. You meant golang builtin [copy](https://pkg.go.dev/builtin#copy) function. I was referring to a function to copy conditions from the api machinery lib. But I guess it is so straightforward. 

the builtin copy only does shallow copy.